### PR TITLE
Editorial: update to modern Web IDL conventions

### PR DESCRIPTION
**Pull request by domenic**
* Uses "getter steps" and "method steps"
* Use "this" instead of "context object"
* Talk about dictionaries as Infra maps
* Minor restructuring of promise-creation/in parallel algorithms for NavigationPreloadManager

***
<a href="https://pr-preview.s3.amazonaws.com/domenic/ServiceWorker/pull/1528.html" title="Last updated on Aug 6, 2020, 8:58 AM UTC (2a02128)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/ServiceWorker/1528/fc328f8...domenic:2a02128.html" title="Last updated on Aug 6, 2020, 8:58 AM UTC (2a02128)">Diff</a>

### CODE REVIEW

File: docs/index.bs [hunk 1]
```diff
@@ -404,7 +404,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
     <section>
       <h4 id="service-worker-url">{{ServiceWorker/scriptURL}}</h4>
 
-      The <dfn attribute for="ServiceWorker"><code>scriptURL</code></dfn> attribute *must* return the [=/service worker=]'s <a lt="URL serializer">serialized</a> [=service worker/script url=].
+      The <dfn attribute for="ServiceWorker"><code>scriptURL</code></dfn> getter steps  are to return the [=/service worker=]'s <a lt="URL serializer">serialized</a> [=service worker/script url=].
```
Comment (jakearchibald) [inline docs/index.bs:407]:
```suggestion
      The <dfn attribute for="ServiceWorker"><code>scriptURL</code></dfn> getter steps are to return the [=/service worker=]'s <a lt="URL serializer">serialized</a> [=service worker/script url=].
```

File: docs/index.bs [hunk 2]
```diff
@@ -704,9 +704,9 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
 
       <dfn attribute for="ServiceWorkerContainer"><code>controller</code></dfn> attribute *must* run these steps:
 
-        1. Let |client| be the <a>context object</a>'s [=ServiceWorkerContainer/service worker client=].
+        1. Let |client| be [=this=]'s [=ServiceWorkerContainer/service worker client=].
         1. If |client|'s [=active service worker=] is null, then return null.
-        1. Return the result of [=getting the service worker object=] that represents |client|'s [=active service worker=] in the [=context object=]'s [=relevant settings object=].
+        1. Return the result of [=getting the service worker object=] that represents |client|'s [=active service worker=] in the [=this=]'s [=relevant settings object=].
```
Comment (jakearchibald) [inline docs/index.bs:709]:
```suggestion
        1. Return the result of [=getting the service worker object=] that represents |client|'s [=active service worker=] in [=this=]'s [=relevant settings object=].
```

File: docs/index.bs [hunk 3]
```diff
@@ -731,24 +731,24 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
...
-        1. Let |scriptURL| be the result of <a lt="URL parser">parsing</a> |scriptURL| with the <a>context object</a>'s <a>relevant settings object</a>'s <a>API base URL</a>.
+        1. Let |client| be [=this=]'s [=ServiceWorkerContainer/service worker client=].
+        1. Let |scriptURL| be the result of <a lt="URL parser">parsing</a> |scriptURL| with [=this=]'s <a>relevant settings object</a>'s <a>API base URL</a>.
         1. Let |scopeURL| be null.
-        1. If |options|.{{RegistrationOptions/scope}} is <a>present</a>, set |scopeURL| to the result of <a lt="URL parser">parsing</a> |options|.{{RegistrationOptions/scope}} with the <a>context object</a>'s <a>relevant settings object</a>'s <a>API base URL</a>.
-        1. Invoke [=Start Register=] with |scopeURL|, |scriptURL|, |p|, |client|, |client|'s <a>creation URL</a>, |options|.{{RegistrationOptions/type}}, and |options|.{{RegistrationOptions/updateViaCache}}.
+        1. If |options|["{{RegistrationOptions/scope}}"] [=map/exists=], set |scopeURL| to the result of <a lt="URL parser">parsing</a> |options|["{{RegistrationOptions/scope}}"] with [=this=]'s <a>relevant settings object</a>'s <a>API base URL</a>.
```
Comment (jakearchibald) [inline docs/index.bs:740]: Ohhh TIL!

File: docs/index.bs [hunk 4]
```diff
@@ -814,7 +814,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
         </tbody>
       </table>
 
-      The first time the <a>context object</a>'s {{ServiceWorkerContainer/onmessage}} IDL attribute is set, its [=ServiceWorkerContainer/client message queue=] *must* be enabled.
+      The first time the {{ServiceWorkerContainer/onmessage}} setter steps are performed, [=this=]'s [=ServiceWorkerContainer/client message queue=] *must* be enabled.
```
Comment (jakearchibald) [inline docs/index.bs:817]:
```suggestion
      The first time the {{ServiceWorkerContainer/onmessage}} setter steps are performed, enable [=this=]'s [=ServiceWorkerContainer/client message queue=].
```

Unrelated change, but eliminates another `*must*`

File: docs/index.bs [hunk 5]
```diff
@@ -900,46 +900,58 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
     <section algorithm="navigation-preload-manager-enable">
       <h4 id="navigation-preload-manager-enable">{{NavigationPreloadManager/enable()}}</h4>
 
-      The <dfn method for="NavigationPreloadManager"><code>enable()</code></dfn> method, when invoked, *must* return a new [=promise=] |promise| and run the following steps [=in parallel=]:
```
Comment (jakearchibald) [inline docs/index.bs:903]: Is this shorthand out of favour now (if it ever was)?

File: docs/index.bs [hunk 6]
```diff
@@ -1854,7 +1863,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
                 1. If |r|'s [=request/method=] is not \`<code>GET</code>\` and |options|.ignoreMethod is false, return [=a promise resolved with=] an empty array.
             1. Else if |request| is a string, then:
                 1. Set |r| to the associated [=Request/request=] of the result of invoking the initial value of {{Request}} as constructor with |request| as its argument. If this [=throws=] an exception, return [=a promise rejected with=] that exception.
-        1. Let |realm| be the [=context object=]'s [=relevant realm=].
+        1. Let |realm| be the [=this=]'s [=relevant realm=].
```
Comment (jakearchibald) [inline docs/index.bs:1866]:
```suggestion
        1. Let |realm| be [=this=]'s [=relevant realm=].
```

File: docs/index.bs [hunk 7]
```diff
@@ -1934,7 +1943,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
                 1. Set |operation|'s [=cache batch operation/response=] to |response|.
                 1. [=list/Append=] |operation| to |operations|.
                 1. Increment |index| by one.
-            1. Let |realm| be the [=context object=]'s [=relevant realm=].
+            1. Let |realm| be the [=this=]'s [=relevant realm=].
```
Comment (jakearchibald) [inline docs/index.bs:1946]:
```suggestion
            1. Let |realm| be [=this=]'s [=relevant realm=].
```

File: docs/index.bs [hunk 8]
```diff
@@ -1978,7 +1987,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
         1. Set |operation|'s [=cache batch operation/request=] to |innerRequest|.
         1. Set |operation|'s [=cache batch operation/response=] to |clonedResponse|.
         1. [=list/Append=] |operation| to |operations|.
-        1. Let |realm| be the [=context object=]'s [=relevant realm=].
+        1. Let |realm| be the [=this=]'s [=relevant realm=].
```
Comment (jakearchibald) [inline docs/index.bs:1990]:
```suggestion
        1. Let |realm| be [=this=]'s [=relevant realm=].
```

File: docs/index.bs [hunk 9]
```diff
@@ -2006,7 +2015,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
         1. Set |operation|'s [=cache batch operation/request=] to |r|.
         1. Set |operation|'s [=cache batch operation/options=] to |options|.
         1. [=list/Append=] |operation| to |operations|.
-        1. Let |realm| be the [=context object=]'s [=relevant realm=].
+        1. Let |realm| be the [=this=]'s [=relevant realm=].
```
Comment (jakearchibald) [inline docs/index.bs:2018]:
```suggestion
        1. Let |realm| be [=this=]'s [=relevant realm=].
```

File: docs/index.bs [hunk 10]
```diff
@@ -2031,7 +2040,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
                 1. If |r|'s [=request/method=] is not \`<code>GET</code>\` and |options|.ignoreMethod is false, return [=a promise resolved with=] an empty array.
             1. Else if |request| is a string, then:
                 1. Set |r| to the associated [=Request/request=] of the result of invoking the initial value of {{Request}} as constructor with |request| as its argument. If this [=throws=] an exception, return [=a promise rejected with=] that exception.
-        1. Let |realm| be the [=context object=]'s [=relevant realm=].
+        1. Let |realm| be the [=this=]'s [=relevant realm=].
```
Comment (jakearchibald) [inline docs/index.bs:2043]:
```suggestion
        1. Let |realm| be [=this=]'s [=relevant realm=].
```

File: docs/index.bs (hunk 5 above)
Comment (domenic) [reply docs/index.bs:903]: Yeah, the reader's eyes skip over the prelude and miss important facts like promise creation or running in parallel.

**Comment by domenic** [comment]
Thanks for fixing up all the things I missed!

**Pull request merged**